Description: actually check the return codes from rsync and other commands.
Author: Kees Cook <kees@ubuntu.com>
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=455082

Index: debmirror-2.4.4/debmirror
===================================================================
--- debmirror-2.4.4.orig/debmirror	2010-04-05 11:40:30.292651025 -0700
+++ debmirror-2.4.4/debmirror	2010-04-05 11:40:35.402651381 -0700
@@ -1280,6 +1280,7 @@
 	    }
 	  }
 	  system ("rsync --no-motd --timeout=$timeout $opt $rsyncremote --include-from=$rsynctempfile --exclude='*' $mirrordir");
+	  die "rsync failed!" if ($? != 0);
 	  close $fh;
 	  unlink $rsynctempfile;
 	  foreach my $dest (@result) {
@@ -1781,7 +1782,10 @@
   if (-f $tofile) {
     unlink($tofile) or die "unlink($tofile) failed: $!";
   }
-  system("$command <$fromfile >$tofile");
+  my $cmd="$command <$fromfile >$tofile";
+  say("Running: $cmd") if ($verbose);
+  system("$cmd");
+  die "Failed: $cmd\n" if ($? != 0);
 }
 
 sub split_dist {
